### PR TITLE
Nz mn delete item

### PR DESCRIPTION
**Pull request by N-Zubko**
## Description

The PR adds a new feature to the app which allows users to delete items from the list. 

We were thinking of adding a message to keep users informed about the deletion result. @michellerenehey suggested adding an alert, which seems better that displaying a line with a message in the parent element. Let us know what you think about it.

## Related Issue

Closes issue #11 

## Acceptance Criteria

- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [X]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
List page before the changes

### After

1. List with Delete buttons
List with Delete buttons

2. Dialog window
Dialog window

3. Confirmation of item deleted
Confirmation of item deleted

4. Item was not deleted
Item was not deleted

## Testing Steps / QA Criteria

1. Checkout `NZ-MN-delete-item` for the latest changes on your local computer and `git pull`, then `npm start`.
2. Make sure you have some items in your list.
3. Click `Delete` button on any item and then choose `OK` in the dialog window. You will see a window confirming that the item has been deleted.
4. Click `Delete` button on any item and then choose `Cancel` in the dialog window. You will see a window informing that the item has not been deleted.

### CODE REVIEW

**Review by Trevor-Rezac — APPROVED** [review]
Everything looks great, really clean code! I wasn't familiar with window.confirm yet so thats neat! My only suggestion is to remove the alert after the item was deleted (line 65 ListItem.jsx). IMO it just seems like an extra step for the user to have to click OK again after confirming they wanted to delete the item. Again, just my opinion though so if the rest of the group want's to keep that in there, totally fine with me!

**Comment by N-Zubko** [comment]
> My only suggestion is to remove the alert after the item was deleted (line 65 ListItem.jsx). IMO it just seems like an extra step for the user to have to click OK again after confirming they wanted to delete the item. 

Thank you for sharing your opinion @Trevor-Rezac ! I totally agree that the extra step can be skipped, but would like to hear other reviewers' opinions

**Review by lindseyindev — APPROVED** [review]
Great work on this PR! I made some non-blocking comments and my opinion on the alert() and I leave it up to you to make any final updates from the feedback here!

File: src/components/ListItem.jsx [hunk 1]
```diff
@@ -65,6 +78,9 @@ export function ListItem({ item, listToken }) {
 				checked={isPurchased}
 			/>
 			<label htmlFor={`${item.id}-${item.name}-checkbox`}>{item.name}</label>
+			<button type="button" className="" onClick={handleDeleteItem}>
```
[the remark this answers is not in the document]
Comment (N-Zubko) [reply src/components/ListItem.jsx:81]: I will remove it, forgot about it, thank you @lindseyindev

**Review by mira-kine — APPROVED** [review]
Nice straightforward and clean code - the window.confirm as mentioned by the others was a good choice and is something I've never seen before. I see nothing that needs to be changed in my opinion. Good work!

**Comment by N-Zubko** [comment]
Commented out the alert() part in the handleDeleteItem function after all reviewers reported on it causing the need to click excessively

**Pull request merged**